internal/pkg/manifest/validate.go

```diff
@@ -1498,6 +1498,21 @@ func (s SidecarConfig) validate() error {
 	}
 	return s.ImageOverride.validate()
 }
+func (s SidecarConfig) validateSideCarImageFields() error {
```

Comment:
Can we simplify the logic by

1. first check if `image` is set. If not error out saying it is a required field;
2. if `image` is a struct, check if either `location` or `build` is set.

This way we'll have clearer error msg for customers and from impl perspective we don't need to define a new error type specifically for this.